### PR TITLE
rewrite collect.js, fixes #23

### LINKED ISSUE #23: Change collect.js to do nothing and have "event hooks"
Instead of just doing `<script src="webalyzer.domain/collect.js" data-domain="mydomain"></script>`
The user would do:

```
<script src="webalyzer.domain/collect.js"></script>
<script>
Webalyzer.configure('mydomain', {
    doSend: function(url) {
        return Math.random() > 0.7;
    }
});
```

### PR DESCRIPTION
**Pull request by peterbe**
@stephaniehobson r?
cc @groovecoder 

We chatted about NOT using the domain and instead using a "key" instead. That's a good idea but it's something we can do separately.

### CODE REVIEW

**Comment by groovecoder** [comment]
Does this conflict with https://github.com/mdn/webalyzer/pull/22 and/or https://github.com/mdn/webalyzer/pull/18 ? Can it wait to rebase onto one or both of those?

**Comment by peterbe** [comment]
> Does this conflict with #22 and/or #18 ? Can it wait to rebase onto one or both of those?

I'm happy to wait and rebase mine.

File: README.md [hunk 1]
```diff
@@ -77,11 +77,42 @@ Use locally
 
 1. Add a `<script>` tag to your page for `collect.js`:
 
-        <script src="http://127.0.0.1:8000/static/collect.js"
-        data-webalyzer="example.com"></script>
+        <script>
+        (function() {
+        var s = document.createElement('script');
+        s.src = 'http://127.0.0.1:8000/static/collect.js';
+        document.head.appendChild(s); s.async=true; s.onload=function() {
```
Comment (stephaniehobson) [inline README.md]: async and onload should be attached before we append the script tag.

File: webalyzer/collector/static/collect.js [hunk 2]
```diff
@@ -67,6 +64,7 @@
 
   function postHTML(html, callback) {
     var data = new FormData();
+    html = options.beforeSendHTML(html, document.location.href);
```
Comment (stephaniehobson) [inline webalyzer/collector/static/collect.js]: When you do the merge make sure this line ends up as the first thing in postHTML (ie before the source_hash check.)

File: README.md [hunk 3]
```diff
@@ -77,11 +77,42 @@ Use locally
...
+You can do this:
+
+        Webalyzer('example.com', {
+             doSend: function(html, url) {
+                 if (getWeather() === 'raining') {
+                     return false;
+                 }
+                 return true;
+             },
+             beforeSendHTML: function(html, url) {
```
Comment (stephaniehobson) [inline README.md:107]: We probably want a beforeSendCSS as well. Not a blocker for this PR though, additional feature.

**Comment by stephaniehobson** [comment]
Not a comprehensive review since I don't have celery working :( I'll need to review again after the rebase anyway, hopefully celery will be cooperating by then.

File: README.md (hunk 1 above)
Comment (peterbe) [inline README.md]: Right you are!

File: webalyzer/collector/static/collect.js (hunk 2 above)
Comment (peterbe) [inline webalyzer/collector/static/collect.js]: taken care of now

File: webalyzer/collector/static/collect.js [hunk 4]
```diff
@@ -1,15 +1,12 @@
...
-  }
+var Webalyzer = function(domain, options) {
+  options = options || {};
+  options.doSend = options.doSend || function(html, url) {
+    return true;
+  };
+  options.beforeSendHTML = options.beforeSendHTML || function(html, url) {
+    return html;
+  };
+  options.server = options.server || 'https://webalyzer.dev/collector/';
```
Comment (peterbe) [inline webalyzer/collector/static/collect.js:9]: Perhaps the default should be the heroku URL? Can you remind me what that was @groovecoder ?

File: webalyzer/collector/urls.py [hunk 5]
```diff
@@ -6,9 +6,9 @@
 urlpatterns = patterns(
     '',
     url(
-        re.compile('^check/(?P<source_type>[\w]+)/'
-                   '(?P<domain>[\w\.]+)/'
-                   '(?P<source_hash>[\-\w\.]+)$'),
+        '^check/(?P<source_type>[\w]+)/'
+        '(?P<domain>[\w\.]+)/'
+        '(?P<source_hash>[\-\w\.]+)$',
```
Comment (peterbe) [inline webalyzer/collector/urls.py:10]: WIthout this change it doesn't match.

File: webalyzer/collector/static/collect.js (hunk 4 above)
Comment (stephaniehobson) [inline webalyzer/collector/static/collect.js:9]: http://webalyzer.herokuapp.com/ right now.

**Comment by stephaniehobson** [comment]
Got it working locally, looks good :)

@groovecoder  You will need to update the sites you have using the snippet!

**Pull request merged**